autoparser/csv_mapping.py

```diff
@@ -0,0 +1,190 @@
+            score += scores["type-match"]
+        if attributes.get("format") == "date":
+            score += (
+                scores["type-match"]
+                if "date_" in str(row.get("source_validation_type", ""))
+                else scores["date-mismatch"]
+            )
+        if (
+            "follow" in row["category"]
+        ):  # de-emphasise followup, usually only required in observation
+            score += scores["is-followup"]
+        words = row["field"].split("_")
+        score += sum(
+            w in row["source_field"] + " " + row["source_field_description"]
+            for w in set(words) - set(stopwords)
+        )
+        return score * row["tf_rank"]
+
+    match_df["score"] = match_df.apply(scorer, axis=1)
+    return match_df.sort_values(["field", "score"], ascending=[True, False])
+
+
+def deep_get(d: dict, keys: str):
+    dc = copy.deepcopy(d)
+    ks = keys.split(".")
+    for k in ks:
+        dc = dc[k]
+    return dc
+
+
+def read_json(file: str) -> dict:
+    with (Path(__file__).parent / file).open() as fp:
+        return json.load(fp)
+
+
+def get_fields(config: dict[str], table: str) -> list[str]:
+    schemas = config.get("schemas", [])
+    use_fields = config.get("use_fields", {})
+    if table not in schemas:
+        raise ValueError(f"Schema not found for table: {table}")
+    if table not in use_fields:
+        return read_json(schemas[table])
+    else:
+        return {
+            "properties": {
+                k: {"category": "observation"}
+                for k in deep_get(read_json(schemas[table]), use_fields[table])
+            }
+        }
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Generate intermediate CSV used by make_toml.py to create TOML"
+    )
+    parser.add_argument("dictionary", help="Data dictionary to use")
+    parser.add_argument(
+        "-o", "--output", help="Name to use for output files", default="isaric"
+    )
+    parser.add_argument(
+        "-n",
+        "--num_matches",
+        help="Number of matches to output for each field",
+        type=int,
+        default=6,
+    )
+    parser.add_argument(
+        "-t", "--tables", help="Only match for tables (comma separated list, no spaces)"
+    )
+    args = parser.parse_args()
+    with open(Path(__file__).parent / "config.json") as fp:
+        config = json.load(fp)
+    # for table in config["schemas"]:
+    table = "subject"
+    tables = args.tables.split(",") if args.tables else config["schemas"].keys()
+    for table in tables:
+        df = matches_redcap(
+            args.dictionary,
+            table,
+            config["column_mappings"],
+            get_fields(config, table),
+            scores=config.get("scores", {}),
+            num_matches=args.num_matches,
+            stopwords=config.get("stopwords", "english"),
+        )
+        df.to_csv(f"{args.output}-{table}.csv", index=False)
```

Comment:
Ok, so this outputs a csv for each table, and not one large csv for all tables combined?

Reply:
Yes, easier to review that way